### PR TITLE
storage: derive `Arbitrary` where possible

### PR DESCRIPTION
**Pull request by petrosagg**
### Motivation

Just some noise reduction. There was no reason for these manual implementations.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by rjobanp — APPROVED** [review]
> +76 −520

😄 

looks good, thanks for cleaning this up!

**Review by teskje — COMMENTED** [review]
I also like the boilerplate reduction, but do you have guidance on one the manual `Arbitrary` impl is needed and when it isn't? I assume it has to do with wanting to use the `BoxedStrategy`?

**Comment by petrosagg** [comment]
> but do you have guidance on one the manual Arbitrary impl is needed and when it isn't?

my plan is to see if CI + Nightly pass without stack overflows and if yes, merge them. I don't have any guidance unfortunately

File: src/mysql-util/src/desc.rs [hunk 1]
```diff
@@ -28,9 +28,11 @@ pub struct MySqlTableDesc {
     /// The index of each column is based on its `ordinal_position`
     /// reported by the information_schema.columns table, which defines
     /// the order of column values when received in a row.
+    #[proptest(strategy = "proptest::collection::vec(any::<MySqlColumnDesc>(), 0..4)")]
```
Comment (guswynn) [inline src/mysql-util/src/desc.rs:31]: til about this!

**Review by guswynn — COMMENTED** [review]
looks good!

I assume nightly failures are unrelated

**Review by guswynn — APPROVED** [review]

**Pull request merged**